### PR TITLE
Unified Distributed Storage API implementation 

### PR DESCRIPTION
**Pull request by LiZhenCheng9527**
…age updates

**What type of PR is this?**
Unified Distributed Storage API Implementation and Dependency Package Follow-Up

**What this PR does / why we need it**:
Complete API design of storage and go dependency package updates

### CODE REVIEW

File: pkg/apis/fleet/v1alpha1/types.go [hunk 1]
```diff
@@ -329,6 +332,161 @@ type BackupStorageLocation struct {
...
+	// extraArgs:
+	//   image:
+	//     pullPolicy: Always
+	//
+	// +optional
+	ExtraArgs apiextensionsv1.JSON `json:"extraArgs,omitempty"`
+}
+
+type DistributedStorage struct {
+	// The path on the host where config and data can be persisted. Must be set.
```
Comment (hzxuzhonghu) [inline pkg/apis/fleet/v1alpha1/types.go:362]: remove the` must be set`

File: pkg/apis/fleet/v1alpha1/types.go [hunk 2]
```diff
@@ -329,6 +332,161 @@ type BackupStorageLocation struct {
...
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// +nullable
+	// +optional
+	StorageDeviceSelection `json:",inline"`
+}
+
+type StorageDeviceSelection struct {
+	// Whether to consume all the storage devices found on a machine
+	// indicating whether all devices found on nodes in the cluster should be automatically consumed by OSDs.
+	//Not recommended unless you have a very controlled environment where you will not risk formatting of devices with existing data.
```
Comment (hzxuzhonghu) [inline pkg/apis/fleet/v1alpha1/types.go:477]: white space

File: pkg/apis/fleet/v1alpha1/types.go [hunk 3]
```diff
@@ -329,6 +332,161 @@ type BackupStorageLocation struct {
...
+	// +optional
+	UseAllDevices *bool `json:"useAllDevices,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/fleet/v1alpha1/types.go:479]: rm

**Comment by Xieql** [comment]
your force push overwrite the origin commit, so reviewers can not get the delta change

File: pkg/apis/fleet/v1alpha1/types.go [hunk 4]
```diff
@@ -394,26 +394,23 @@ type MonSpec struct {
...
 	// +optional
-	Labels rookv1.LabelsSpec `json:"labels,omitempty"`
+	Labels map[string]string `json:"labels,omitempty"`
 
 	// The placement-related configuration to pass to kubernetes (affinity, node selector, tolerations).
 	// +kubebuilder:pruning:PreserveUnknownFields
 	// +nullable
 	// +optional
-	Placement rookv1.PlacementSpec `json:"placement,omitempty"`
+	Placement map[string]string `json:"placement,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/fleet/v1alpha1/types.go:413]: This is not right, 

we need such struct
https://github.com/rook/rook/blob/ffaf4a1b3fcea3e0cabb6436328f643ad59407f8/pkg/apis/ceph.rook.io/v1/types.go#L2763C24-L2779

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**